nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core.Test/Update/UpdateWorkerTestBase.cs

```diff
@@ -201,6 +223,42 @@ public abstract class UpdateWorkerTestBase : TestBase
         AssertContainsFiles(expectedResult, actualResult);
     }
 
+    public static async Task MockNuGetPackagesInDirectory(MockNuGetPackage[]? packages, string temporaryDirectory)
```

Comment:
This is where the fake NuGet packages are written to a local feed and an appropriate `NuGet.Config` is created.

Reply:
If no `packages` parameter is specified, no local feed and no `NuGet.Config` file will be created, which will still allow tests to hit the network if they're still required.  As soon as a test sets `packages: [...]` then all network access it cut off.  I did this to allow for the most flexibility, but also as a way to ensure that a test that specifies `packages` won't accidentally pull something from `api.nuget.org`.

Reply:
I wonder if this method should be extracted somewhere more central, since it isn't Updater specific.

Reply:
I made it `public` so I could use it elsewhere (in fact, the `discover` tests use it), I just didn't have a good place to put it and I wanted to get this huge PR off of my box :)